approov-service/src/main/java/io/approov/service/okhttp/ApproovException.java

```diff
@@ -17,17 +17,141 @@
+public static final int ERROR_TOKEN_FETCH_POOR_NETWORK = 103;
+public static final int ERROR_TOKEN_FETCH_MITM_DETECTED = 104;
+public static final int ERROR_TOKEN_FETCH_NO_SERVICE = 105;
+public static final int ERROR_TOKEN_FETCH_UNKNOWN_URL = 106;
+public static final int ERROR_TOKEN_FETCH_UNPROTECTED_URL = 107;
+public static final int ERROR_TOKEN_FETCH_TIMEOUT = 108;
+public static final int ERROR_TOKEN_FETCH_UNKNOWN_FAILURE = 199;
+// ==========================================
+
+private final int errorCode;
     /**
      * Constructs an exception due to an Approov error.
      *
      * @param message is the basic information about the exception cause
+     * @Depriciated use ApproovException(int errorCode, String message) instead.
      */
     public ApproovException(String message) {
-        super(message);
+        this(ERROR_LEGACY, message);
+    }
+
+    /**
+     * Constructs an exception due to an Approov error while wrapping the cause
+     *
+     * @param message is the basic information about the exception cause
+     * @param cause is the underlying cause of the exception
+     * @Depriciated use ApproovException(int errorCode, String message, Throwable cause) instead.
+     */
+    public ApproovException(String message, Throwable cause) {
+        this(ERROR_LEGACY, message, cause);
+    }
+
+    /**
+     * Constructs an exception with a specific error code and message.
+     *
+     * @param errorCode provides a machine readable reason for the failure
+     * @param message is the basic information about the exception cause
+     */
+    public ApproovException(int errorCode, String message) {
+        this(errorCode, message, null);
+    }
+
+    /**
+     * Constructs an exception with an error code and cause.
+     *
+     * @param errorCode provides a machine readable reason for the failure
+     * @param message is the basic information about the exception cause
+     * @param cause is the underlying cause of the exception
+     */
+    public ApproovException(int errorCode, String message, Throwable cause) {
+        super(message, cause);
+        this.errorCode = errorCode;
     }
+
+    /**
+     * Constructs an exception with a TokenFetchStatus and message.
+     *
+     * @param status TokenFetchStatus that triggered the error
+     * @param message is the basic information about the exception cause
+     * */
+    public ApproovException(Approov.TokenFetchStatus status, String message) {
+        super(message, null);
+        this.errorCode = mapTokenFetchStatus(status);
+    }
+    /**
+     * Constructs an exception with TokenFetchStatus, message and Throwable cause .
+     *
+     * @param status TokenFetchStatus that triggered the error
+     * @param message is the basic information about the exception cause
+     * @param cause is the underlying cause of the exception
+     */
+    public ApproovException(Approov.TokenFetchStatus status, String message, Throwable cause) {
```

Comment:
Is this one used? It doesn't seem like it would ever be possible because either you get the status  or an exception not both. I think you should remove it.